### PR TITLE
Zero line feature for the Spectra widget

### PR DESCRIPTION
**Pull request by borondics**
Added an option to the Spectra widget to toggle a horizontal zero-line that can be sometimes useful for inspecting PCA loadings. The line can be toggled from the dropdown menu or by pressing 0.

Partially implements https://github.com/Quasars/orange-spectroscopy/issues/544

### CODE REVIEW

File: orangecontrib/spectroscopy/widgets/owspectra.py [hunk 1]
```diff
@@ -669,6 +669,7 @@ class CurvePlot(QWidget, OWComponent, SelectionGroupMixin):
     invertX = Setting(False)
     viewtype = Setting(INDIVIDUAL)
     show_grid = Setting(False)
+    show_zeroLine = Setting(False)
```
Comment (markotoplak) [inline orangecontrib/spectroscopy/widgets/owspectra.py:672]: Dont use camelCase.

**Comment by stuart-cls** [comment]
Works for me, looks nice with 2nd derivatives too! Doesn't clash with show grid setting either. Although the list of visualization tweaks that poor little popup has to hold is getting long...

**Comment by borondics** [comment]
Actually, you have a point for the Grid. I should turn off the zero line if the grid is on. 

For the menu - yeah... I really hope someone will implement a GMail-like overlay menu. Or we could add submenus...

**Comment by stuart-cls** [comment]
I think it's better to leave zero line / grid settings unconnected: if we're going to expose both might as well let the user choose.

**Comment by markotoplak** [comment]
@stuart-cls, @borondics, do you think that this is really needed?

From a functionality perspective, the grid covers the use cases I envision. I would avoid adding such features unless they are very good reasons for it. This feature is nicely written and not problematic for maintenance, but I afraid of feature creep in the UI.

**Comment by borondics** [comment]
This was a request during a workshop in relation to looking at second derivatives. I agree that the grid takes care of it though. Maybe what we could do is to add this zero-line in the preprocessors automatically and not expose it as a menu item. However then we still have the problem with the Spectra or Hyperspectra widgets and the behaviours of the different spectral plots would diverge... What do you think?

**Comment by markotoplak** [comment]
I think the grid is enough and that we should not cater to every request. We probably forgot about the grid at the time - if we showed it then, they would likely be satisfied.

**Comment by borondics** [comment]
Does the grid always go through 0? If it does I think we can drop this zero-line feature.

**Pull request closed without merging**